packages/graphql/src/schema/scalars/Time.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GraphQLError, GraphQLScalarType, Kind } from "graphql";
+import neo4j from "neo4j-driver";
+
+export const RFC_3339_REGEX = /^(?<hour>[01]\d|2[0-3]):(?<minute>[0-5]\d):(?<second>[0-5]\d)(\.(?<fraction>\d{1}(?:\d{0,8})))?((?:[Zz])|((?<offsetDirection>[-|+])(?<offsetHour>[01]\d|2[0-3]):(?<offsetMinute>[0-5]\d)))?$/;
+
+export const validateTime = (value: any) => {
+    if (typeof value !== "string") {
+        throw new TypeError(`Value must be of type string: ${value}`);
+    }
+
+    if (!RFC_3339_REGEX.test(value)) {
+        throw new TypeError(`Value must be formatted as Time: ${value}`);
+    }
+
+    return value;
+};
+
+export const parseTime = (value: any) => {
+    const validatedValue = validateTime(value);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const { hour, minute, second, fraction, offsetDirection, offsetHour, offsetMinute } = RFC_3339_REGEX.exec(
+        validatedValue
+    )!.groups!;
```

Comment:
As far as I know, RegEx matching is quite expensive, so doing it once with `test` in `validateTime` and then again with `exec` here is a bit wasteful. See my suggestion below for potentially getting rid of `validateTime` and performing the whole lot in `parseTime` - it also means one less non-null assertion also.

```suggestion
export const parseTime = (value: any) => {
    if (typeof value !== "string") {
        throw new TypeError(`Value must be of type string: ${value}`);
    }
    
    const match = RFC_3339_REGEX.exec(value);
    
    if (!match) {
        throw new TypeError(`Value must be formatted as Time: ${value}`);
    }
    
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const { hour, minute, second, fraction, offsetDirection, offsetHour, offsetMinute } = match.groups!;
```

Reply:
Great point.